tmt/steps/prepare/artifact/providers/repository.py

```diff
@@ -0,0 +1,131 @@
+"""
+Artifact provider for repository files.
+"""
+
+from collections.abc import Iterator
+from re import Pattern
+from shlex import quote
+from urllib.parse import unquote, urlparse
+
+import tmt.log
+from tmt.steps.prepare.artifact.providers import (
+    ArtifactProvider,
+    DownloadError,
+)
+from tmt.steps.prepare.artifact.providers.koji import RpmArtifactInfo
+from tmt.steps.provision import Guest
+from tmt.utils import GeneralError, Path, ShellScript
+
+
+class RepositoryFile:
+    """
+    A helper class representing a repository .repo file from a URL.
+    """
+
+    def __init__(self, url: str) -> None:
+        # The constructor also serves as a validator for the URL format
+        try:
+            result = urlparse(url)
+            if not result.scheme or not result.netloc:
+                raise ValueError
+        except ValueError:
+            raise GeneralError(f"Invalid URL format for .repo file: '{url}'.")
+        self._url = url
+
+    @property
+    def url(self) -> str:
+        """The URL of the .repo file."""
+        return self._url
+
+    @property
+    def filename(self) -> str:
+        """A suitable filename extracted from the URL path."""
+        path = urlparse(self.url).path
```

Comment:
Since `urlparse()` is called in `__init__()`, maybe we could save its output, e.g. in `self._url_components`, and just use `self._url_components.path`.

Reply:
Addressed in https://github.com/teemtee/tmt/pull/4018/commits/309d42757a5a75ebec1f8cdafcca7aca135b357c